scrapy/http/request/__init__.py

```diff
@@ -132,6 +136,15 @@ def __init__(
             dict(cb_kwargs) if cb_kwargs else None
         )
         self.flags: List[str] = [] if flags is None else list(flags)
+        # Check for mutable objects in cb_kwargs if JOBDIR is set
+        if settings and 'JOBDIR' in settings:
```

Comment:
`JOBDIR` is always in settings though, via `default_settings`?

Reply:
My implementation checks if JOBDIR is set to a non-None value to ensure that the warning is only issued when JOBDIR is explicitly set by the user.

Reply:
> My implementation checks if JOBDIR is set to a non-None value

Where?